src/mlmc.py

```diff
@@ -0,0 +1,181 @@
+import numpy as np
+from src.mc_level import Level
+
+
+class MLMC:
+    """
+    Multi level monte carlo method
+    """
+    def __init__(self, number_of_levels, n_fine, n_coarse, sim):
+        """
+        :param number_of_levels:           Number of levels
+        :param n_fine:      Maximum steps of one simulation
+        :param n_coarse:    Minimum steps of one simulation
+        :param sim:         Instance of object Simulation
+        """
+
+        self.simulation = sim
+        # Initialization of variables
+        self._levels = []
+        self.target_time = None
+        self.target_variance = None
+        self.number_of_levels = number_of_levels
+        self.n_fine = n_fine
+        self.n_coarse = n_coarse
+        self._number_of_simulations = None
+        self.moments_function = None
+        self.moments_object = None
+        self.index = 0
+
+    @property
+    def levels(self):
+        """
+        Get monte carlo method levels
+        """
+        return self._levels
+
+    @levels.setter
+    def levels(self, levels):
+        if not isinstance(levels, list):
+            raise TypeError("Levels must be list")
+        self._levels = levels
+
+    @property
+    def number_of_simulations(self):
+        """
+        Number of simulations
+        """
+        return self._number_of_simulations
+
+    @number_of_simulations.setter
+    def number_of_simulations(self, num_of_sim):
+        if len(num_of_sim) != self.number_of_levels:
+            raise ValueError("Number of simulations must be list and the length of list must be same"
+                             " as the number of levels")
+
+        self._number_of_simulations = num_of_sim
+
+    def monte_carlo(self, optimization_type, value):
```

Comment:
Remove this method:
- call directly estimate_n_samples_XY, and pass there target time or target variance as parameter;
  estimate methods should take maximum of new estimate and existing estimate of n_samples for individual levels.
- make levels in constructor
- user will call 'refill_samples' explicitely

This allows